lib/components/primitive-components/Group/Group_doInitialSchematicTraceRender/applyTracesFromSolverOutput.ts

```diff
@@ -46,6 +57,37 @@ export function applyTracesFromSolverOutput(args: {
           const existing = db.schematic_port.get(schPid)
           if (existing) db.schematic_port.update(schPid, { is_connected: true })
         }
+
+        // Attempt to derive the connectivity key from the matched source_trace
+        const source_trace = source_trace_id
+          ? db.source_trace.get(source_trace_id)
+          : undefined
+        if (source_trace?.subcircuit_connectivity_map_key) {
+          subcircuit_connectivity_map_key =
+            source_trace.subcircuit_connectivity_map_key
+        } else {
+          // Fallback: infer from the connected ports' source_ports
+          const src_A = pA ? schPortIdToSourcePortId.get(pA) : undefined
+          const src_B = pB ? schPortIdToSourcePortId.get(pB) : undefined
+          const source_port_A = src_A ? db.source_port.get(src_A) : undefined
+          const source_port_B = src_B ? db.source_port.get(src_B) : undefined
+          const subcircuit_connectivity_map_key_A =
+            source_port_A?.subcircuit_connectivity_map_key
+          const subcircuit_connectivity_map_key_B =
+            source_port_B?.subcircuit_connectivity_map_key
+          if (
+            subcircuit_connectivity_map_key_A &&
+            subcircuit_connectivity_map_key_B &&
+            subcircuit_connectivity_map_key_A ===
+              subcircuit_connectivity_map_key_B
+          )
+            subcircuit_connectivity_map_key = subcircuit_connectivity_map_key_A
+          else
+            subcircuit_connectivity_map_key =
+              subcircuit_connectivity_map_key_A ||
+              subcircuit_connectivity_map_key_B ||
+              undefined
```

Comment:
This seems wrong. The schematic trace solver should have this information already,